CAL-51 Add ability to specify timeout on IOR file retrieval.

troymohl opened the pull request:
#### What does this PR do?
Add ability to specify timeout on IOR file retrieval.
Add additional IOR/CORBA debugging for host/port of remote ORB.
#### Who is reviewing it?
@jaymcnallie @bdeining @lcrosenbu @michaelmenousek 
#### How should this be tested?
Set logging for NSILI  to DEBUG:  log:set DEBUG org.codice.alliance.nsili
Add NSILI Federated Source
Verify option exists to set timeout for IOR retrieval
Verify that IOR details are logged at debug level
#### Any background context you want to provide?
This will provide additional logging to help debug firewall and connectivity problems for STANAG-4559 sources.
The IOR.java file is reused from BBN OpenMap. Checkstyle is configured to ignore it for headers and Jacoco ignores it for unit testing metrics.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/CAL-51
#### Screenshots (if appropriate)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

bdeining commented on catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/IOR.java:
```diff
@@ -0,0 +1,334 @@
+            try {
+                ior[iorIndex] = (byte) ((hexByteToInt(hex[hexIndex]) << 4) + (hexByteToInt(hex[hexIndex + 1])));
+            } catch (NumberFormatException e) {
+                System.err.println("Index: " + hexIndex);
```
If we are allowed to modify this class, I suggest using our logging vs system.out / err

troymohl commented:
Updated PR with the following:
1. Changed IOR.java to write everything to the PrintWriter instead of System.err, System.out. 
2. Added ability to control Corba timeouts (in addition to cxf)
3. Changed Nsili to be less restrictive on the @Produces

troymohl commented:
Another update to convert the IOR debug parsing to use the LOGGER directly for output logging. Removed the main that won't be used in our configuration.

bdeining commented on catalog/nsili/catalog-nsili-source/src/main/resources/OSGI-INF/blueprint/blueprint.xml:
```diff
@@ -60,6 +60,7 @@
             <property name="cxfUsername" value=""/>
             <property name="cxfPassword" value=""/>
+            <property name="cxfTimeout" value="60"/>
             <property name="pollInterval" value="5"/>
```
Should corbaTimeout be on both the Federated and Connected sources?

troymohl replied:
Good catch. Missed that in the blueprint.xml

bdeining commented on catalog/nsili/catalog-nsili-source/src/main/java/org/codice/alliance/nsili/source/NsiliSource.java:
```diff
@@ -184,6 +188,10 @@
 
     private String cxfPassword;
 
+    private int cxfTimeout = 60;
```
nit: I don't think these need to be set if a value is specified in blueprint
